### PR TITLE
Remove SbUser from SbStorage APIs

### PR DESCRIPTION
**Pull request by gbournou**
As part of a broader effort to eliminate the entire `starboard/user.h`, removing the concept of a "user" from the `SbStorage` APIs defined in `starboard/storage.h`. Storage will now be user-agnostic, since almost all platforms don't actually rely on having per-device-user storage.

b/271930936

### CODE REVIEW

File: starboard/nplb/storage_open_record_test.cc [hunk 1]
```diff
@@ -1,38 +0,0 @@
...
-// limitations under the License.
-
-// Sunny Day cases tested in the read/write tests.
-
-#include "starboard/common/storage.h"
-#include "starboard/nplb/file_helpers.h"
-#include "starboard/user.h"
-#include "testing/gtest/include/gtest/gtest.h"
-
-namespace starboard {
```
Comment (kaidokert) [inline starboard/nplb/storage_open_record_test.cc:22]: General patterns with those ( unfortunately ) is also that we keep the tests and #ifdef them out. Until the time the deprecated code falls out of Starboard support window in 3 years. ( i know )

File: starboard/nplb/storage_delete_record_test.cc [hunk 2]
```diff
@@ -1,37 +0,0 @@
...
-// limitations under the License.
-
-// Sunny Day cases tested in the read/write tests.
-
-#include "starboard/common/storage.h"
-#include "starboard/nplb/file_helpers.h"
-#include "starboard/user.h"
-#include "testing/gtest/include/gtest/gtest.h"
-
-namespace starboard {
```
Comment (kaidokert) [inline starboard/nplb/storage_delete_record_test.cc:22]: General patterns with those ( unfortunately ) is also that we keep the tests and #ifdef them out. Until the time the deprecated code falls out of Starboard support window in 3 years. ( i know )

File: starboard/common/storage.cc [hunk 3]
```diff
@@ -17,24 +17,35 @@
...
+StorageRecord::StorageRecord(SbUser user)
+    : user_(user), record_(kSbStorageInvalidRecord) {
   Initialize();
 }
 
 StorageRecord::StorageRecord(SbUser user, const char* name)
     : user_(user), name_(name), record_(kSbStorageInvalidRecord) {
   Initialize();
 }
+#endif
```
Comment (kaidokert) [inline starboard/common/storage.cc:48]: Tiny nitpick: we generally add a comment after endif, like `#endif // SB_API_VERSION < 16` - makes the long ifdef blocks easier to grok. That's here and elsewhere below.

**Review by kaidokert — APPROVED** [review]
Overall looks ok - but please keep the deprecated tests ( they'll still run in sbversion-15 and sbversion-14 builds ) and a nitpick about ending #endifs

**Review by y4vor — APPROVED** [review]

**Pull request merged**